src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.cs

```diff
@@ -17,6 +17,8 @@ internal sealed partial class AsynchronousOperationListener : IAsynchronousOpera
 
         private readonly string _featureName;
         private readonly HashSet<TaskCompletionSource<bool>> _pendingTasks = new HashSet<TaskCompletionSource<bool>>();
+        private CancellationTokenSource _blockedOnCompletionTokenSource = new CancellationTokenSource();
```

Comment:
either here or in constructor? rather than both?

Reply:
Fixed